dbms/programs/server/Server.cpp

```diff
@@ -166,6 +192,22 @@ int Server::main(const std::vector<std::string> & /*args*/)
     std::string path = getCanonicalPath(config().getString("path", DBMS_DEFAULT_PATH));
     std::string default_database = config().getString("default_database", "default");
 
+    /// Check that the process' user id matches the owner of the data.
```

Comment:
Why do you need to check the owner of the data? Isn't in better just to check if folder is writale by effective user_id? 

So, for example, why can't the owner be userX and folder have 777 permissions and is 100% writable by ClickHouse? (of course, setting 777 is not a brilliant idea, but I don't see the reasons to limit that especially for some testing purposes).

Reply:
The idea is to disallow creation of files with root's ownership if the server is run under the root by a mistake. Such files mess with a correct run under a dedicated user later. Since any folder is writable by the root a check that it is writable would not help to prevent the mistake. See details in the corresponding [task](https://trello.com/c/kbHeEudQ/46-%D0%BF%D1%80%D0%BE%D0%B2%D0%B5%D1%80%D0%BA%D0%B0-%D0%BF%D1%80%D0%B8-%D1%81%D1%82%D0%B0%D1%80%D1%82%D0%B5-%D1%87%D1%82%D0%BE-%D1%81%D0%B5%D1%80%D0%B2%D0%B5%D1%80-%D0%B7%D0%B0%D0%BF%D1%83%D1%89%D0%B5%D0%BD-%D0%B8%D0%B7-%D0%BF%D0%BE%D0%B4-%D0%BF%D0%BE%D0%B4%D1%85%D0%BE%D0%B4%D1%8F%D1%89%D0%B5%D0%B3%D0%BE-%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D0%B5%D0%BB%D1%8F).

Reply:
Ok, but may be that behaviour should be adjustable, like with flag `--skip_owner_check`. As i can imagine situations when owner and effective user differ.

Reply:
We can throw and refuse to start only in the case of root, but just print a warning in other cases.

> So, for example, why can't the owner be userX and folder have 777 permissions and is 100% writable by clickhouse user? (of course, setting 777 is not a brilliant idea, but I don't see the reasons to limit that, especially for some testing purposes).

We should check data directory. Example: `/var/lib/clickhouse/data` instead of `/var/lib/clickhouse`

Reply:
> We can throw and refuse to start only in the case of root, but just print a warning in other cases.
I've implemented this solution.